cobra/core/reaction.py

```diff
@@ -739,6 +739,14 @@ def add_metabolites(self, metabolites_to_add, combine=True,
         _id_to_metabolites = dict([(x.id, x) for x in self._metabolites])
 
         for metabolite, coefficient in iteritems(metabolites_to_add):
+
+            # Make sure metabolites being added belong to the same model, or
+            # else copy them.
+            if isinstance(metabolite, Metabolite):
+                if ((metabolite.model is not None) &
+                        (metabolite.model != self._model)):
```

Comment:
Pardon my nitpicking. Can you turn the `&` into an `and`, please. In the second comparison we're not interested in model (in-)equality but identity, right? So `metabolite.model is not self._model` seems more idiomatic to me.